addons/addon-base-raas/packages/base-raas-services/lib/data-egress/data-egress-service.js

```diff
@@ -124,15 +137,104 @@ class DataEgressService extends Service {
         },
       ],
     };
+    const memberAccountId = await this.getMemberAccountId(requestContext, environment.id);
 
-    const environmentScService = await this.service('environmentScService');
-    const environmentScEntity = await environmentScService.mustFind(requestContext, { id: environment.id });
-    const memberAccount = await environmentScService.getMemberAccount(requestContext, environmentScEntity);
-    await this.addEgressStoreBucketPolicy(requestContext, egressStore, memberAccount.accountId);
+    const lockService = await this.service('lockService');
+    const lockId = `bucket-policy-access-${bucketName}`;
+    await lockService.tryWriteLockAndRun({ id: lockId }, async () => {
+      await this.addEgressStoreBucketPolicy(requestContext, egressStore, memberAccountId);
+    });
 
     return egressStore;
   }
 
+  async terminateEgressStore(requestContext, environmentId) {
+    const enableEgressStore = this.settings.get(settingKeys.enableEgressStore);
+    const by = _.get(requestContext, 'principalIdentifier.uid');
```

Comment:
Can this API be called by anyone? I am assuming there should be a check to see either the caller is an SWB admin or owns the environment?

Reply:
Fix applied in commit: 3917dc0ed4b0fa6bb7aafbd67d45cd7846a58922